shell/platform/embedder/embedder_mapping.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder_mapping.h"
+#include "flutter/shell/platform/embedder/embedder_struct_macros.h"
+
+namespace flutter {
+
+class EmbedderMapping final : public fml::Mapping {
```

Comment:
This class is unnecessary and can be replaced by `fml::NonOwnedMapping`. You may construct the `ReleaseProc` using captures. So, something like:

```
release_proc = [destruction_callback, user_data](const uint8_t* data, size_t size {
  destruction_callback(user_data);
};
```